app-tasks/rf/src/rf/commands/ingest_scene.py

```diff
@@ -75,27 +77,46 @@ def save_ingest_def_to_s3(scene_id, ignore_previous=False):
     return ingest_definition.s3_uri, ingest_definition.id
 
 
-def launch_spark_ingest_job(ingest_def_uri, ingest_def_id, scene_id):
+def launch_spark_ingest_job(ingest_def_uri, ingest_def_id, scene_id, local=False):
     """Launch ingest job and wait for success/failure
 
     Args:
         ingest_def_uri (str): S3 URI for location of ingest definition
         ingest_def_id (str): ID for ingest definition
         scene_id (str): ID for scene to be ingested
+        local (bool): whether to execute ingest as a spark local job
 
     Returns:
         bool
     """
 
     logger.info('Launching Spark ingest job with ingest definition %s for scene %s',
                 ingest_def_uri, scene_id)
-    cluster_id = get_cluster_id()
-    emr_response = execute_ingest_emr_job(scene_id, ingest_def_uri, ingest_def_id, cluster_id)
-    logger.info('Launched Spark ingest job for %s with ingest ID %s. Waiting for status changes.',
-                scene_id, ingest_def_id)
-    step_id = emr_response['StepIds'][0]
-    is_success = wait_for_emr_success(step_id, cluster_id)
-    return is_success
+    if local:
+        return execute_local_ingest_job(scene_id, ingest_def_uri)
+    else:
+        cluster_id = get_cluster_id()
+        emr_response = execute_ingest_emr_job(scene_id, ingest_def_uri, ingest_def_id, cluster_id)
+        logger.info('Launched Spark ingest job for %s with ingest ID %s. Waiting for status changes.',
+                    scene_id, ingest_def_id)
+        step_id = emr_response['StepIds'][0]
+        is_success = wait_for_emr_success(step_id, cluster_id)
+        return is_success
+
+
+def execute_local_ingest_job(scene_id, ingest_s3_uri):
+    """Execute ingest job locally with a spark local cluster"""
+
+    command = ['spark-submit',
+               '--master', 'local[32]', '--driver-memory', '48g',
```

Comment:
Probably going to want to parameterize these two later since they seem closely coupled with similar parameters in the job definition. They line up now, so makes sense to keep as is, but I suspect this will be one of the first things we want to tweak when things start going wrong.

Reply:
Good point, I'll make the edit

Reply:
Addressed in https://github.com/raster-foundry/raster-foundry/pull/2806/commits/a20c1074c295534f1d7164cab3914fec46ba0d18